### PR TITLE
Main person

### PR DESCRIPTION
**Pull request by maria-salmasian**
added the project person with packaging

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,24 @@
 /.idea/
 *.iml
-/out/
```
Comment (nsimonyan) [inline .gitignore:3]: Please make sure don't write all exclusions one by one use generic regex

File: MainPerson/src/com/beyond/person/core/service/implementation/DancerImpl.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    public void learn() {
+        System.out.println( dancer.getName() + " has a dancing Education");
+
+    }
+
+    /**
+     * all methods of dancer in one
+     */
+    @Override
+    public void all() {
```
Comment (nsimonyan) [inline MainPerson/src/com/beyond/person/core/service/implementation/DancerImpl.java:49]: Try to make all available for base class but at the same time add in each exact impl specific behavior

File: MainPerson/src/com/beyond/person/core/service/interfaces/BasePersonAction.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.beyond.person.core.service.interfaces;
+
+import com.beyond.person.core.model.BasePerson;
+
+public interface BasePersonAction {
+    /**
+     * method walk same for all
+     * @param person
+     */
+    default void walk(BasePerson person){
```
Comment (nsimonyan) [inline MainPerson/src/com/beyond/person/core/service/interfaces/BasePersonAction.java:10]: Why do you need default ?

**Review by nsimonyan — CHANGES_REQUESTED** [review]

File: MainPerson/src/com/beyond/person/core/service/interfaces/BasePersonAction.java (hunk 3 above)
Comment (maria-salmasian) [reply MainPerson/src/com/beyond/person/core/service/interfaces/BasePersonAction.java:10]: for it to work the same way for all the people, as i dont have abstract class basepersonImpl, thats why i used this

File: MainPerson/src/com/beyond/person/core/service/implementation/DancerImpl.java (hunk 2 above)
Comment (maria-salmasian) [reply MainPerson/src/com/beyond/person/core/service/implementation/DancerImpl.java:49]: it is available for the base class, it is a method in basepersonAction, and the other interfaces take this method from the basePersonAction, and each implement a specific behavior

File: .gitignore (hunk 1 above)
Comment (maria-salmasian) [reply .gitignore:3]: deleted all the production files and added only *production to git ignore

**Review by nsimonyan — APPROVED** [review]

**Pull request merged**